src/main/java/kafka/Parser.java

```diff
@@ -0,0 +1,93 @@
+package kafka;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDateTime;
+
+public class Parser {
+
+    public static void parseCommand(String userInput, TaskList taskList, Storage storage, Ui ui) throws KafkaException, IOException, DateTimeParseException {
+        String[] splitInput = userInput.trim().split(" ", 2);
+
+        if (splitInput[0] == null) {
+            return;
+        }
+
+        String command = splitInput[0].toLowerCase();
+        String arguments = splitInput.length > 1 ? splitInput[1] : "";
+
+        switch (command) {
```

Comment:
Wow! This is a really long switch case. Could this be broken up into smaller parts? Maybe try creating a method for each case! This would improve code readability!